includes/Story_Post_Type.php

```diff
@@ -315,6 +316,37 @@ public function change_default_title( $data ) {
 		return $data;
 	}
 
+	/**
+	 * Handles redirects to the post type archive.
+	 *
+	 * @since 1.12.0
+	 *
+	 * @param bool      $bypass Pass-through of the pre_handle_404 filter value.
+	 * @param \WP_Query $query The WP_Query object.
+	 * @return bool Whether to pass-through or not.
+	 */
+	public function redirect_post_type_archive_urls( $bypass, $query ) {
+		global $wp_rewrite;
+
+		if ( $bypass || ! is_string( $this->get_has_archive() ) || ( ! $wp_rewrite instanceof \WP_Rewrite || ! $wp_rewrite->using_permalinks() ) ) {
+			return $bypass;
+		}
+
+		// 'pagename' is for most permalink types, name is for when the %postname% is used as a top-level field.
+		if ( self::REWRITE_SLUG === $query->get( 'pagename' ) || self::REWRITE_SLUG === $query->get( 'name' ) ) {
+			$redirect_url = get_post_type_archive_link( self::POST_TYPE_SLUG );
```

Comment:
Could there be an infinite loop if the page slug is `/web-stories/` and thus `get_post_type_archive_link( self::POST_TYPE_SLUG )` returns `https://example.com/web-stories/`?

Reply:
I don't think so. The check ```! is_string( $this->get_has_archive() )``` means that check to see if archive is a page. If the page url is `/web-stories/`, then it will not 404, meaning this could will never run.